rust/src/storage/file/mod.rs

```diff
@@ -92,10 +92,16 @@ impl StorageBackend for FileStorageBackend {
         let readdir = ReadDirStream::new(fs::read_dir(path).await?);
 
         Ok(Box::pin(readdir.err_into().and_then(|entry| async move {
-            Ok(ObjectMeta {
-                path: String::from(entry.path().to_str().unwrap()),
-                modified: DateTime::from(entry.metadata().await.unwrap().modified().unwrap()),
-            })
+            let path = String::from(entry.path().to_str().unwrap());
+            // We check if the file is a temporary file created to prepare a commit.
+            // Getting meta data is another system call on some platforms and the file
+            // may have been removed in the meantime ...
+            let modified = if path.contains("_commit_") {
```

Comment:
Good catch on this race condition!

I think there is still a race here because other files could be deleted through vacuuming as well. What do you think about matching on the metadata error and ignore the path if error matches `Os { code: 2, kind: NotFound }`?

Reply:
ha, nvm, @wjones127 beat me to it :D